packages/eslint-plugin/docs/rules/no-circular-import.mdx

```diff
@@ -0,0 +1,47 @@
+---
+description: 'Disallow the use of import module that result in circular imports.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-circular-import** for documentation.
+
+This rule disallows the use of import module that result in circular imports except for the type-only imports.
```

Comment:
Add explanation about what benefit it has over `import/no-cycle` (i.e. uses the TS resolver and is faster)